pkg/apis/rook.io/v1/types.go

```diff
@@ -217,6 +217,9 @@ type VolumeSource struct {
 	// CrushDeviceClass represents the crush device class for an OSD
 	// +optional
 	CrushDeviceClass string `json:"crushDeviceClass,omitempty"`
+	// CrushInitialWeight represents initial OSD weight (float, TiB units)
+	// +optional
+	CrushInitialWeight string `json:"crushInitialWeight,omitempty"`
```

Comment:
Since it's optional please make it a pointer instead.

Reply:
Also, please investigate `resource.Quantity` and see if it makes sense here, it could allow use to enter more arbitrary 
 value that can be easily converted internally. If not, please add an pattern validation like:

```
// +kubebuilder:validation:Pattern=`^[0-9]*([T]i[B])$`
```

If the expectation is TiB

Reply:
With strings I would hope we could treat the empty string as not set instead of using a pointer.

Reply:
Indeed, using empty string as "not-set" feels more intuitive than using a pointer, and as far as I can tell, it is also a common pattern in other places in the code. Using ```resource.Quantity``` could have been better, but unfortunately, it has rather limited meaning withing kubernetes terminology (cpu/memory; extending it to capacity is not a trivial effort).